### PR TITLE
Add support for ledger nano sim to CI

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This PR adds CI support for a ledger nano simulator and tests for ledger nano.

## Links to any relevant issues

- #387 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

File: .github/actions/ledger-nano/action.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: 'ledger-nano-setup'
+description: 'Setup a simulated ledger nano instance'
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout ledger shimmer app files
+      uses: actions/checkout@v3
+      with:
+        repository: iotaledger/ledger-shimmer-app
+        path: ledger-shimmer-app
```
Comment (shufps) [inline .github/actions/ledger-nano/action.yml]: ledger-shimmer-app is obsolete. It was all merged into ledger-iota-app and for the `build.sh` you need to specify the variant with `-v shimmer`

Comment (DaughterOfMars) [reply .github/actions/ledger-nano/action.yml]: Is that all that you think needs to be done?

Comment (DaughterOfMars) [reply .github/actions/ledger-nano/action.yml]: Still getting `DeviceNotFound` in the test.

File: sdk/src/client/secret/ledger_nano.rs [hunk 2]
```diff
@@ -109,6 +108,8 @@ impl From<iota_ledger_nano::api::errors::APIError> for Error {
 pub struct LedgerSecretManager {
     /// Specifies if a real Ledger hardware is used or only a simulator is used.
     pub is_simulator: bool,
+    /// Specifies whether the wallet should be in non-interactive mode.
+    pub non_interactive: bool,
```
Comment (thibault-martinez) [inline sdk/src/client/secret/ledger_nano.rs:112]: Changelog :trollface:

File: .github/actions/ledger-nano/action.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: 'ledger-nano-setup'
+description: 'Setup a simulated ledger nano instance'
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout ledger shimmer app files
+      uses: actions/checkout@v3
+      with:
+        repository: iotaledger/ledger-iota-app
+        ref: fix/build-without-it-flag
```
Comment (thibault-martinez) [inline .github/actions/ledger-nano/action.yml]: Was that temporary or... ?

Comment (DaughterOfMars) [reply .github/actions/ledger-nano/action.yml]: The branch is, yes. Not sure if it's merged yet.

Comment (DaughterOfMars) [reply .github/actions/ledger-nano/action.yml]: It is not merged yet.

Comment (thibault-martinez) [reply .github/actions/ledger-nano/action.yml]: Maybe we ask Thomas to merge it before?

Comment (thibault-martinez) [reply .github/actions/ledger-nano/action.yml]: Was merged

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**